upb/decode.c

```diff
@@ -483,6 +484,7 @@ static const char *decode_enum_packed(upb_decstate *d, const char *ptr,
   while (!decode_isdone(d, &ptr)) {
     wireval elem;
     ptr = decode_varint64(d, ptr, &elem.uint64_val);
+    decode_munge(field->descriptortype, &elem);
```

Comment:
Actually it might be better to factor out the int32/enum logic from `decode_munge()` into a separate function, called from both `decode_munge()` and here. We want to avoid having to re-check `type` here, especially since the code is a complete no-op on little-endian platforms.

Reply:
Was https://github.com/protocolbuffers/upb/pull/435/commits/1c2921f0d4ac83af749ab4a714c9743092dad0b1 what you had mind?

Reply:
Yep that's perfect, just fix the indentation and we should be good.